read_buffer/src/lib.rs

```diff
@@ -181,6 +188,77 @@ impl Database {
             .sum()
     }
 
+    /// Returns the schema for the specified columns in the provided table, for
+    /// the specified partition key and chunks within that partition.
+    pub fn table_schema<'a>(
+        &self,
+        partition_key: &str,
+        table_name: &'a str,
+        chunk_ids: &[u32],
+        select_columns: ColumnSelection<'a>,
+    ) -> Result<Schema> {
+        let partition = self
+            .partitions
+            .get(partition_key)
+            .context(PartitionNotFound { key: partition_key })?;
+
+        let schema_builder =
+            chunk_ids
+                .iter()
+                .try_fold(SchemaMerger::new(), |builder, chunk_id| {
+                    let chunk = partition
+                        .chunks
+                        .get(chunk_id)
+                        .context(ChunkNotFound { id: *chunk_id })?;
+
+                    let result_schema = chunk.table_schema(table_name, &select_columns);
+
+                    let builder = if let Some(result_schema) = result_schema {
+                        let schema = result_schema
+                            .try_into()
+                            .context(BuildingSchema { chunk_ids })?;
+
+                        builder
+                            .merge(schema)
+                            .context(BuildingSchema { chunk_ids })?
+                    } else {
+                        builder
+                    };
+                    Ok(builder)
+                })?;
+
+        let schema = schema_builder
+            .build()
+            .context(BuildingSchema { chunk_ids })?;
+
+        // Ensure the order of the output columns is as requested
+        let schema = if matches!(select_columns, ColumnSelection::All) {
+            schema.sort_fields_by_name()
+        } else {
+            schema
+        };
+
+        Ok(schema)
+    }
+
+    /// returns true if the table exists in at least one of the specified chunks
+    pub fn has_table(&self, partition_key: &str, table_name: &str, chunk_ids: &[u32]) -> bool {
+        if let Some(partition) = self.partitions.get(partition_key) {
+            for chunk_id in chunk_ids {
+                let chunk_has_table = partition
+                    .chunks
+                    .get(chunk_id)
+                    .map(|chunk| chunk.has_table(table_name))
+                    .unwrap_or(false);
+
+                if chunk_has_table {
+                    return true;
+                }
+            }
```

Comment:
I think this slightly shorter alternative does the same:

```suggestion
            return chunk_ids
                .iter()
                .any(|chunk_id| match partition.chunks.get(chunk_id) {
                    Some(chunk) => chunk.has_table(table_name),
                    None => false,
                });
```